data/tutorials/ds_05_seq.md

```diff
@@ -0,0 +1,313 @@
+```
+Here, `close_in` will never be called over the input channel opened on
+`README.md`.
+
+## Sequences Are Functions
+
+Although this looks like a possible way to define the [Fibonacci
+sequence](https://en.wikipedia.org/wiki/Fibonacci_number):
+```ocaml
+# let rec fibs m n = Seq.cons m (fibs n (n + m));;
+val fibs : int -> int -> int Seq.t = <fun>
+```
+It actually isn't. It's a non-ending recursion which blows away the stack.
+```
+# fibs 0 1;;
+Stack overflow during evaluation (looping recursion?).
+```
+This definition is behaving as expected:
+```ocaml
+# let rec fibs m n () = Seq.Cons (m, fibs n (n + m));;
+val fibs : int -> int -> int Seq.t = <fun>
+```
+It can be used to produce some Fibonacci numbers:
+```ocaml
+# fibs 0 1 |> Seq.take 10 |> List.of_seq;;
+- : int list = [0; 1; 1; 2; 3; 5; 8; 13; 21; 34]
+```
+Why is it so? The key difference lies in the recursive call `fibs n (n + m)`. In
+the former definition, the application is complete because `fibs` is provided
+all the arguments it expects. In the latter definition, the application is
+partial because the `()` argument is missing. Since evaluation is
+[eager](https://en.wikipedia.org/wiki/Evaluation_strategy#Eager_evaluation) in
+OCaml, in the former case, evaluation of the recursive call is triggered and a
+non-terminating looping occurs. In contrast, in the latter case, the partially
+applied function is immediately returned as a
+[closure](https://en.wikipedia.org/wiki/Closure_(computer_programming)).
+
+Sequences are functions, as stated by their type:
+```ocaml
+# #show Seq.t;;
+type 'a t = unit -> 'a Seq.node
+```
+Functions working with sequences must be written accordingly.
+* Sequence consumer: partially applied function parameter
+* Sequence producer: partially applied function result
+
+When code dealing with sequences does not behave as expected like if it is
+crashing or hanging, there's a fair chance a mistake like in the first
+definition of `fibs` was made.
+
+## Sequences for Conversions
+
+Throughout the standard library, sequences are used as a bridge to perform
+conversions between many datatypes. For instance, here are the signatures of
+some of those functions:
+* Lists
+  ```ocaml
+  val List.of_seq : 'a list -> 'a Seq.t
+  val List.to_seq : 'a Seq.t -> 'a list
+  ```
+* Arrays
+  ```ocaml
+  val Array.of_seq : 'a array -> 'a Seq.t
+  val Array.to_seq : 'a Seq.t -> 'a array
+  ```
+* Strings
+  ```ocaml
+  val String.of_seq : string -> char Seq.t
+  val String.to_seq : char Seq.t -> string
+  ```
+Similar functions are also provided for sets, maps, hash tables (`Hashtbl`) and
+others (except `Seq`, obviously). When implementing a datatype module, it is
+advised to expose `to_seq` and `of_seq` functions.
+
+## Miscellaneous Considerations
+
+There are a couple of related libraries, all providing means to handle large
+flows of data:
+
+* Rizo I [Streaming](/p/streaming)
+* Gabriel Radanne [Iter](/p/iter)
```

Comment:
I'm pretty sure that the library was built by Simon Cruanes and Gabriel Radanne. So I suggest: 

```diff
-   * Gabriel Radanne [Iter](/p/iter)
+  * Simon Cruanes and Gabriel Radanne [Iter](/p/iter)
```